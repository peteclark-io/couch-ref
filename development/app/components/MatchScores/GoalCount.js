'use strict'

import React from 'react';
import styles from './styles.css';

const GoalCount = React.createClass({

   propTypes: {
      'goals': React.PropTypes.number
   },

   render: function() {
      return (
         <span className={styles.goals} ref={node => (this.root = node)}>{this.props.goals}</span>
      );
   }
});

export default GoalCount;
