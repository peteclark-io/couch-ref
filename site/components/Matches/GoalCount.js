'use strict'

import React from 'react';
import styles from './styles.css';

class GoalCount extends React.Component {

  static propTypes = {
    'goals': React.PropTypes.number
  };

  componentDidMount() {
    window.componentHandler.upgradeElement(this.root);
  }

  componentWillUnmount() {
    window.componentHandler.downgradeElements(this.root);
  }

  render() {
    return (
      <span className={styles.goals} ref={node => (this.root = node)}>{this.props.goals}</span>
    );
  }

}

export default GoalCount;
