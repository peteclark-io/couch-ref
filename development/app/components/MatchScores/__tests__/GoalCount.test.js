'use strict';

import React from 'react';
import GoalCount from '../GoalCount';
import renderer from 'react-test-renderer';

it('Renders ok.', () => {
   const rendered = renderer.create(
      <GoalCount goals={4} />
   );
   expect(rendered.toJSON()).toMatchSnapshot();
});
