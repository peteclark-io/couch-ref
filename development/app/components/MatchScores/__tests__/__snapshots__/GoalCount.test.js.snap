exports[`test Renders ok. 1`] = `
<span
  className="goals">
  4
</span>
`;
